### ISSUE TITLE
`carthage update` hangs on build

### ISSUE THREAD

**Issue by cybertk** [Carthage/Carthage#522]
Given ` github "cybertk/JSONModel" "carthage"` in **Cartfile**, when `carthage update`, it hangs up like following.

```
$ carthage update
*** Fetching JSONModel
*** Checking out JSONModel at "04ef2ed9d5898e38ba30cf4299d964eba32a18aa"
*** xcodebuild output can be found in /var/folders/ft/d3401yq51xn4_8yyh18kqfsr0000gq/T/carthage-xcodebuild.40edEO.log
```

```
$ carthage version
0.7.3
```

**Comment by Goles** [comment]
I think this has been addressed by 0.7.4 beta ( you can download the beta 2 from a Package installer )

**Comment by cybertk** [comment]
@Goles Could you tell more details about this issue? What is the root cause? I will try to verify this on 0.7.4 beta

**Comment by neilpa** [comment]
It's not easily summarized but take a look at  #445, #455 and #474 if you want more details.

**Comment by jspahrsummers** [comment]
I've tagged [0.7.4](https://github.com/Carthage/Carthage/releases/tag/0.7.4), which should fix all the hanging and crashing issues that we know about. The update should [become available in Homebrew](https://github.com/Homebrew/homebrew/pull/40450) soon.

**Comment by Goles** [comment]
Perfect @jspahrsummers  :+1:

**Comment by cybertk** [comment]
@jspahrsummers My issue does not fixed in **0.7.4**, it still hangs on `carthage build` for `cybertk/JSONModel`

**Comment by cybertk** [comment]
It can be easily reproduced by following commands.

```
git clone --depth 1 -b carthage https://github.com/cybertk/JSONModel
cd JSONModel
carthage build --no-skip-current
```

It will hangs with

```
*** xcodebuild output can be found in /var/folders/ft/d3401yq51xn4_8yyh18kqfsr0000gq/T/carthage-xcodebuild.B7BNEd.log
```

**Comment by schwa** [comment]
Still seeing hangs here myself… I'll capture a spin dump when i see it next.

Less hangs for sure than pre 0.7.4

**Comment by ikesyo** [comment]
@cybertk I tested as what you said, and confirmed the followings:

1. Carthage picks up the shared scheme `JSONModel` from `JSONModel.xcodeproj`, then tries to build the scheme using workspace `JSONModelDemos.xcworkspace`. Using a workspace over a project is intended behavior for building some dependencies for the scheme.
2. In the workspace or projects included in the workspace (`JSONModelDemo_iOS.xcodeproj`, `JSONModelDemo_OSX.xcodeproj`) have no any shared schemes. And also at just after cloning the repository, there are no user-specific schemes for the workspace or projects because they are not opend in Xcode.

  In that case, calling `xcodebuild -workspace JSONModelDemos.xcworkspace -scheme JSONModel  -showBuildSettings` directly also hangs. Because `xcodebuild` hangs, Carthage also hangs at that time.

To work around the case, I think you could do which of the followings:

1. Remove the two demo projects and the workspace, and add targets in demo projects directly to `JSONModel.xcodeproj`.
2. Add `JSONModel.xcodeproj` to `JSONModelDemos.xcworkspace`, then configure your demo projects using the built framework, not adding the library source files to the demo projects.

**Comment by cybertk** [comment]
Thanks @ikesyo. I tried workaround #2, it worked for me. But I did not configure demo projects using the built framework. I only add `JSONModel.xcodeproj` to `JSONModelDemos.xcworkspace`

**Comment by ikesyo** [comment]
Closed by #1120.
